src/osd/PG.cc

```diff
@@ -2037,9 +2037,8 @@ bool PG::queue_scrub()
     state_set(PG_STATE_DEEP_SCRUB);
     scrubber.must_deep_scrub = false;
   }
-  if (scrubber.must_repair) {
+  if (scrubber.must_repair || scrubber.auto_repair) {
     state_set(PG_STATE_REPAIR);
-    scrubber.must_repair = false;
```

Comment:
I'd rather keep must_scrub, must_deepscrub, and must_repair functioning consistently with each other.  All you need to do is make sure that auto_repair is set to false when doing a must_repair, so that later if auto_repair is true you know it can only be an auto_repair.